<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>984</width>
    <height>811</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CBI</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources.qrc">
    <normaloff>:/logger.png</normaloff>:/logger.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QPushButton" name="can1_startbn">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>350</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Старт</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="can1_stopbn">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Стоп</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_6">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::MinimumExpanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="can1">
        <property name="text">
         <string>CAN1</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="can2">
        <property name="text">
         <string>CAN2</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QMdiArea" name="mdi">
      <property name="baseSize">
       <size>
        <width>800</width>
        <height>600</height>
       </size>
      </property>
      <property name="viewMode">
       <enum>QMdiArea::TabbedView</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>984</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Устройство</string>
    </property>
    <widget class="QMenu" name="menu_3">
     <property name="title">
      <string>Настройка интерфейсов</string>
     </property>
     <addaction name="action_CAN_1"/>
     <addaction name="action_CAN_2"/>
    </widget>
    <addaction name="devconnect"/>
    <addaction name="devdisconnect"/>
    <addaction name="reboot"/>
    <addaction name="separator"/>
    <addaction name="time_sync"/>
    <addaction name="firmware_update"/>
    <addaction name="separator"/>
    <addaction name="actionPeriphery"/>
    <addaction name="actionCrane"/>
    <addaction name="menu_3"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>Файл</string>
    </property>
    <widget class="QMenu" name="menu_4">
     <property name="title">
      <string>Конвертировать</string>
     </property>
     <addaction name="action_Hex"/>
     <addaction name="actionASC"/>
    </widget>
    <addaction name="fileopen"/>
    <addaction name="filesave"/>
    <addaction name="separator"/>
    <addaction name="action_dec_txt"/>
    <addaction name="action_dec_txt_can2"/>
    <addaction name="separator"/>
    <addaction name="action_dec_txt_can1"/>
    <addaction name="action_dec_txt_can2_2"/>
    <addaction name="separator"/>
    <addaction name="menu_4"/>
    <addaction name="separator"/>
    <addaction name="quit"/>
   </widget>
   <addaction name="menu_2"/>
   <addaction name="menu"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="dockWidget">
   <property name="features">
    <set>QDockWidget::NoDockWidgetFeatures</set>
   </property>
   <property name="windowTitle">
    <string>Отправка</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <layout class="QGridLayout" name="gridLayout_6">
       <item row="0" column="1">
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="9">
        <widget class="QLabel" name="label_25">
         <property name="font">
          <font>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>данные</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="13">
        <widget class="QLineEdit" name="tx_can0_repeat">
         <property name="maximumSize">
          <size>
           <width>30</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>1</string>
         </property>
        </widget>
       </item>
       <item row="0" column="7">
        <widget class="QLabel" name="label_23">
         <property name="font">
          <font>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>длина данных</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="8">
        <widget class="QComboBox" name="tx_can0_len">
         <property name="maximumSize">
          <size>
           <width>30</width>
           <height>16777215</height>
          </size>
         </property>
         <item>
          <property name="text">
           <string>8</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>7</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>6</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>5</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>4</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>3</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>2</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>1</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QComboBox" name="tx_can0_frametype">
         <property name="maximumSize">
          <size>
           <width>40</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="currentIndex">
          <number>1</number>
         </property>
         <item>
          <property name="text">
           <string>SFF</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>EFF</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="0" column="12">
        <widget class="QLabel" name="label_27">
         <property name="font">
          <font>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>повтор</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QLabel" name="label_21">
         <property name="font">
          <font>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="layoutDirection">
          <enum>Qt::LeftToRight</enum>
         </property>
         <property name="text">
          <string>кадр</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="15">
        <widget class="QLineEdit" name="tx_can0_delay">
         <property name="maximumSize">
          <size>
           <width>40</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item row="0" column="11">
        <widget class="QLineEdit" name="tx_can0_data">
         <property name="maximumSize">
          <size>
           <width>130</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>00 00 00 00 00 00 00 00</string>
         </property>
         <property name="maxLength">
          <number>23</number>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QCheckBox" name="tx_can0_check">
         <property name="minimumSize">
          <size>
           <width>200</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>CAN 1</string>
         </property>
        </widget>
       </item>
       <item row="0" column="16">
        <widget class="Line" name="line_9">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="17">
        <spacer name="horizontalSpacer_5">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>60</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="14">
        <widget class="QLabel" name="label_29">
         <property name="font">
          <font>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>задержка, мс</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="4">
        <widget class="QLabel" name="label_31">
         <property name="maximumSize">
          <size>
           <width>12</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>0x</string>
         </property>
        </widget>
       </item>
       <item row="0" column="10">
        <widget class="QLabel" name="label_33">
         <property name="maximumSize">
          <size>
           <width>12</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>0x</string>
         </property>
        </widget>
       </item>
       <item row="0" column="5">
        <widget class="QLineEdit" name="tx_can0_frameid">
         <property name="maximumSize">
          <size>
           <width>70</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>00 00 00 00</string>
         </property>
         <property name="maxLength">
          <number>11</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="6">
        <widget class="QLabel" name="label_24">
         <property name="font">
          <font>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>длина данных</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QComboBox" name="tx_can1_frametype">
         <property name="maximumSize">
          <size>
           <width>40</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="currentIndex">
          <number>1</number>
         </property>
         <item>
          <property name="text">
           <string>SFF</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>EFF</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="0" column="12">
        <widget class="QLineEdit" name="tx_can1_repeat">
         <property name="maximumSize">
          <size>
           <width>30</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>1</string>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QLabel" name="label_22">
         <property name="font">
          <font>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>кадр</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="11">
        <widget class="QLabel" name="label_28">
         <property name="font">
          <font>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>повтор</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="7">
        <widget class="QComboBox" name="tx_can1_len">
         <property name="maximumSize">
          <size>
           <width>30</width>
           <height>16777215</height>
          </size>
         </property>
         <item>
          <property name="text">
           <string>8</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>7</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>6</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>5</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>4</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>3</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>2</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>1</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="0" column="8">
        <widget class="QLabel" name="label_26">
         <property name="font">
          <font>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>данные</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="14">
        <widget class="QLineEdit" name="tx_can1_delay">
         <property name="maximumSize">
          <size>
           <width>40</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item row="0" column="5">
        <widget class="QLineEdit" name="tx_can1_frameid">
         <property name="maximumSize">
          <size>
           <width>70</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>00 00 00 00</string>
         </property>
         <property name="maxLength">
          <number>11</number>
         </property>
        </widget>
       </item>
       <item row="0" column="10">
        <widget class="QLineEdit" name="tx_can1_data">
         <property name="maximumSize">
          <size>
           <width>130</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>00 00 00 00 00 00 00 00</string>
         </property>
         <property name="maxLength">
          <number>23</number>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QCheckBox" name="tx_can1_check">
         <property name="minimumSize">
          <size>
           <width>200</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>CAN 2</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="Line" name="line_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="13">
        <widget class="QLabel" name="label_30">
         <property name="font">
          <font>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>задержка, мс</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="15">
        <widget class="Line" name="line_8">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="16">
        <spacer name="horizontalSpacer_4">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>60</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="4">
        <widget class="QLabel" name="label_32">
         <property name="maximumSize">
          <size>
           <width>12</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>0x</string>
         </property>
        </widget>
       </item>
       <item row="0" column="9">
        <widget class="QLabel" name="label_34">
         <property name="maximumSize">
          <size>
           <width>12</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>0x</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout_12">
       <item row="2" column="0">
        <widget class="Line" name="line_5">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QPushButton" name="pushButton">
         <property name="text">
          <string>Очистить вывод</string>
         </property>
        </widget>
       </item>
       <item row="1" column="4">
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="transmitbn">
         <property name="text">
          <string>Отправить</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QPushButton" name="canceltxbn">
         <property name="text">
          <string>Отмена</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="devconnect">
   <property name="text">
    <string>Подключиться</string>
   </property>
  </action>
  <action name="devdisconnect">
   <property name="text">
    <string>Отключиться</string>
   </property>
  </action>
  <action name="fileopen">
   <property name="text">
    <string>Открыть</string>
   </property>
  </action>
  <action name="filesave">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Сохранить</string>
   </property>
  </action>
  <action name="quit">
   <property name="text">
    <string>Выход</string>
   </property>
  </action>
  <action name="time_sync">
   <property name="text">
    <string>Синхронизация времени</string>
   </property>
  </action>
  <action name="action_2">
   <property name="text">
    <string>безрамочные</string>
   </property>
  </action>
  <action name="firmware_update">
   <property name="text">
    <string>Проверка обновления</string>
   </property>
  </action>
  <action name="action_dec_txt">
   <property name="text">
    <string>Сохранить в dec txt (can1)</string>
   </property>
  </action>
  <action name="action_dec_txt_can2">
   <property name="text">
    <string>Сохранить в dec txt (can2)</string>
   </property>
  </action>
  <action name="action_dec_txt_can1">
   <property name="text">
    <string>Экспортировать файлы в dec txt (can1)</string>
   </property>
  </action>
  <action name="action_dec_txt_can2_2">
   <property name="text">
    <string>Экспортировать файлы в dec txt (can2)</string>
   </property>
  </action>
  <action name="action_CAN_1">
   <property name="text">
    <string>Настройка CAN 1</string>
   </property>
  </action>
  <action name="action_CAN_2">
   <property name="text">
    <string>Настройка CAN 2</string>
   </property>
  </action>
  <action name="convertFiles">
   <property name="text">
    <string>Конвертировать файлы в hex</string>
   </property>
  </action>
  <action name="reboot">
   <property name="text">
    <string>Перезагрузить</string>
   </property>
  </action>
  <action name="action_ASCII">
   <property name="text">
    <string>Конвертировать файлы в asc</string>
   </property>
  </action>
  <action name="action_Hex">
   <property name="text">
    <string>в HEX</string>
   </property>
  </action>
  <action name="actionASC">
   <property name="text">
    <string>в ASC</string>
   </property>
  </action>
  <action name="actionInf">
   <property name="text">
    <string>Идентификационная информация</string>
   </property>
  </action>
  <action name="actionCrane">
   <property name="text">
    <string>Информация о кране</string>
   </property>
  </action>
  <action name="actionPeriphery">
   <property name="text">
    <string>Периферия</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>transmitbn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>transmitbn()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>175</x>
     <y>904</y>
    </hint>
    <hint type="destinationlabel">
     <x>-5</x>
     <y>625</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>canceltxbn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>canceltxbn()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>261</x>
     <y>904</y>
    </hint>
    <hint type="destinationlabel">
     <x>-2</x>
     <y>580</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>can1_startbn()</slot>
  <slot>can1_stopbn()</slot>
  <slot>can1_cfgbn()</slot>
  <slot>transmitbn()</slot>
  <slot>canceltxbn()</slot>
  <slot>clear_windows()</slot>
 </slots>
</ui>
